File: Data Structures & Algorithms/serialize-and-deserialize-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Codec {
    /**
     * Encodes a tree to a single string.
     *
     * @param {TreeNode} root
     * @return {string}
     */
    serialize(root) {
        
        if(root===null) return ''
        const res=[]

        const dfs=(current)=>
        {
            if(current===null)
            {
               return res.push('null')
                
            }
            res.push(current.val)
            dfs(current.left)
            dfs(current.right)
        }
        dfs(root)
        return res.join(',')
    }

    /**
     * Decodes your encoded data to tree.
     *
     * @param {string} data
     * @return {TreeNode}
     */
    deserialize(data) {
         if (data === '') return null;  // Handle the edge case of an empty tree
       let i=0
       const bt=data.split(',')
       const dfs=()=>
       {
        if(bt[i]==='null'){
            i++
            return null
        }
        const node= new TreeNode(parseInt(bt[i++]))
        node.left=dfs()
        node.right=dfs()
        return node
       }
       return dfs()
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.go
func maxSlidingWindow(nums []int, k int) []int {
    n,left,right:=len(nums),0,0
	q,res:=[]int{},[]int{}

	for right < n {
		if len(q)>0 && q[0] <left {
			q=q[1:]
		}

		for len(q) > 0 && nums[q[len(q)-1]] < nums[right] {
			q=q[:len(q)-1]
		}

		q=append(q,right)

		if right-left+1==k{
			res=append(res,nums[q[0]])
			left+=1
		}
		right+=1
	}
	return res
}
